File: Data Structures & Algorithms/rotate-array/submission-0.java
class Solution {
    public void rotate(int[] nums, int k) {
        int tmp = 0;
        for (int count=0;count<k;count++){
            int l=0, r=nums.length-1;
            while(l<r){
                tmp=nums[l];
                nums[l]=nums[r];
                nums[r]=tmp;
                l+=1;
            }
        }
    }
}

File: Data Structures & Algorithms/sqrtx/submission-0.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l,r=0,x
        # 0 9 - 4
        # 
        while l<=r:
            m=r+(l-r)//2
            tmp=m*m
            if tmp==x:
                return m
            elif tmp<x:
                l=m+1
            elif tmp>x:
                r=m-1
        
        if l*l>x:
            return l-1
        return l
